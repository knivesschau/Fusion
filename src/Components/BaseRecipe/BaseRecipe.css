/* mobile styles first, min 320px */
#base-title {
    font-size: 1.2rem;
    text-shadow: 3px 2px 2px white;
    margin-bottom: -.5em;
}

#ingredients-title, #steps-title {
    text-decoration: underline;
    text-shadow: 3px 2px 2px white;
    margin-bottom: -.5em;
}

#steps-title {
    margin-top: 1em;
}

#starting-cuisine {
    font-size: 1.10rem;
    text-shadow: 3px 2px 2px white;
    margin-top: .45em;
    margin-bottom: -.5em;
}

#base-steps, #base-ingredients {
    font-weight: 500;
}

#base-steps {
    padding-left: 1em;
}

#starting-steps {
    text-align: left;
    margin: 1em;
}

#starting-ingredients {
    margin: .5em;
}

#base-ingredients {
    text-align: left;
    padding-left: .5em;
    margin: .65em;
    margin-top: 1em;
}

#hand-mixer-icon {
    height: 45%;
    width: 45%;
    margin-top: .5em;
    margin-bottom: -.25em;
}

/* tablet and desktop styles */ 
@media only screen and (min-width: 768px) {
    #base-title {
        font-size: 2rem;
        margin-top: .25em;
        margin-bottom: .25em;
    }

    #base-ingredients, #base-steps {
        font-size: 1.35rem;
    }

    #base-ingredients {
        display: inline-block;
        text-align: center;
        line-height: 1.5em;
        margin-bottom: -1em
    }

    #starting-ingredients {
        text-align: left;
    }

    #starting-cuisine {
        font-size: 1.5rem;
        margin-top: .5em;
    }

    #hand-mixer-icon {
        height: 35%;
        width: 35%;
    }
}

@media only screen and (min-width: 1200px) {
    #base-title {
        font-size: 2.5rem;
    }

    #base-ingredients, #base-steps {
        font-size: 1.5rem;
    }
    
    #hand-mixer-icon {
        height: 25%;
        width: 25%;
    }
}