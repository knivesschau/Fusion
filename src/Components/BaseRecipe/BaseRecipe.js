import React, { Component } from 'react';
import fusionContext from '../../fusionContext';
import handmixer from '../../images/hand-mixer.png';
import './BaseRecipe.css';

export default class BaseRecipe extends Component {
    static contextType = fusionContext; 

    render() {

        const {base_name, cuisine_name, ingredients, steps} = this.props; 

        // convert recipe ingredient data into presentable format on client
        const convertIngredients = Object.values({ingredients}).toString();
        const convertSteps = Object.values({steps}).toString();

        // convert recipe step data into presentable format on client
        const starter_ingredients = convertIngredients.split('\n');
        const starter_steps = convertSteps.split('\n');

        return (
            <>
            <section className="Base_Recipe">
                
                <div className="Base_RecipeInfo">
                    <img src={handmixer} id="hand-mixer-icon" alt="hand mixer"/>
                    <h2 id="base-title">{base_name}</h2>
                    <h3 id="starting-cuisine">Culinary Style: {cuisine_name}</h3>
                </div>

                <div className="Base_Ingredients">
                        <h4 id="ingredients-title">Ingredients</h4>
                        
                        <ul id="base-ingredients">
                            {starter_ingredients.map((ingredient, i) => {
                                return (
                                    <li id="starting-ingredients" key={i}>{ingredient}</li>
                                );
                            })}
                        </ul>
                    </div>
                
                    <div className="Steps_Section">
                        <h4 id="steps-title">Steps</h4>
                        
                        <ol id="base-steps">
                            {starter_steps.map((step, i) => {
                                return (
                                    <li id="starting-steps" key={i}>{step}</li>
                                );
                            })}
                        </ol>
                    </div>

            </section>
            </>
        );
    };
};