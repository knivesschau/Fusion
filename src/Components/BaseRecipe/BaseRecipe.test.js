import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import BaseRecipe from './BaseRecipe';

describe ('BaseRecipe Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BaseRecipe/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete BaseRecipe Component and all nested Components', () => {
        const wrapper = shallow(<BaseRecipe/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});