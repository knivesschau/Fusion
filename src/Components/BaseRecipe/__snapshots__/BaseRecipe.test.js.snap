// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseRecipe Component Renders the complete BaseRecipe Component and all nested Components 1`] = `
<Fragment>
  <section
    className="Base_Recipe"
  >
    <div
      className="Base_RecipeInfo"
    >
      <img
        alt="hand mixer"
        id="hand-mixer-icon"
        src="hand-mixer.png"
      />
      <h2
        id="base-title"
      />
      <h3
        id="starting-cuisine"
      >
        Culinary Style: 
      </h3>
    </div>
    <div
      className="Base_Ingredients"
    >
      <h4
        id="ingredients-title"
      >
        Ingredients
      </h4>
      <ul
        id="base-ingredients"
      >
        <li
          id="starting-ingredients"
          key="0"
        />
      </ul>
    </div>
    <div
      className="Steps_Section"
    >
      <h4
        id="steps-title"
      >
        Steps
      </h4>
      <ol
        id="base-steps"
      >
        <li
          id="starting-steps"
          key="0"
        />
      </ol>
    </div>
  </section>
</Fragment>
`;
